cmake/SetupCodeChecks.cmake

```diff
@@ -57,6 +57,111 @@ if(UNCRUSTIFY_FOUND)
     add_dependencies(style uncrustify_style)
 endif()
 
+##------------------------------------------------------------------------------
+## blt_add_code_checks( BASE_NAME           <Base name used for created targets>
+##                      SOURCES             [source1 [source2 ...]]
+##                      C_FILE_EXTS         [ext1 [ext2 ...]]
+##                      F_FILE_EXTS         [ext1 [ext2 ...]]
+##                      UNCRUSTIFY_CFG_FILE <path to uncrustify config file>)
+##
+## This macro adds all enabled code check targets for the given SOURCES. It
+## filters based on file extensions.
+##
+## C_FILE_EXTS is an optional list of file extensions to filter out the C/C++ SOURCES.
+## Otherwise it defaults to: ".cpp" ".hpp" ".cxx" ".hxx" ".cc" ".c" ".h" ".hh"
+##
```

Comment:
Can we make it easy for a user to extend this list vs provide their own?

The case I am thinking about is something like  an ".inc" file as c that we want to check, but we don't want to have to re-add all the default c file cases.

Reply:
If we append to the list instead of replacing it, we wouldn't be able to remove extensions from the list.  Though I'm not sure why anyone would want to do that.